drivers/serial/uart_mcux_lpuart.c

```diff
@@ -36,6 +39,7 @@ struct lpuart_dma_config {
 
 struct mcux_lpuart_config {
 	LPUART_Type *base;
+	const struct device *parent_dev;
```

Comment:
just an observation, we could not build in this field if the CONFIG_NXP_LP_FLEXCOMM is not enabled, to save a few words of flash (just one for each flexcomm uart, so its not that important, ie not a blocking suggestion)